test_runner/regress/test_proxy.py

```diff
@@ -628,7 +628,7 @@ def test_sql_over_http_connection_cancel(static_proxy: NeonProxy):
     assert res["command"] == "INSERT", "HTTP query should insert"
     assert res["rowCount"] == 1, "HTTP query should insert"
 
-    res = static_proxy.http_query(query, [0, 1], user="http", password="http", expected_code=400)
+    res = static_proxy.http_query(query, [0, 1], user="http", password="http", expected_code=500)
```

Comment:
this is debatable, but I think under the hood it's `SqlOverHttpError::Postgres` and it's derived from `tokio_postgres::Error` I guess it would be hard to distinguish what's exactly the issue. Fixes:
```
E   AssertionError: response: {'message': 'duplicate key value violates unique constraint "test_table_pkey"', 'code': '23505', 'detail': 'Key (id)=(1) already exists.', 'hint': None, 'position': None, 'internalPosition': None, 'internalQuery': None, 'severity': 'ERROR', 'where': None, 'table': 'test_table', 'column': None, 'schema': 'public', 'dataType': None, 'constraint': 'test_table_pkey', 'file': 'nbtinsert.c', 'line': '666', 'routine': '_bt_check_unique'}
E   assert 500 == 400 
E    +  where 500 = <Response [500]>.status_code
```